src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -60,7 +91,18 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public List<String> getTopThreeWeatherCondition7(){
-        return null;
+
+        Map countByRoadConditions = getCountByRoadSurfaceCondition7();
+        List<String> weatherConditions= new ArrayList<String>();
+        weatherConditions.addAll(countByRoadConditions.keySet());
+        Collections.sort(weatherConditions, new Comparator<String>() {
+                public int compare(String w1, String w2) {
+                    Long l1 = (Long) countByRoadConditions.get(w1);
```

Comment:
minor comment about naming - w1 and w2 can be replaced with weatherLeft and weatherRight, which will again make more sense for new-joiners. 

Always try to use more descriptive names, believe me - it's always good :)